build/Build.TestCalamariFlavourProject.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using JetBrains.Annotations;
+using Nuke.Common;
+using Nuke.Common.IO;
+using Nuke.Common.Tools.DotNet;
+using Serilog;
+
+namespace Calamari.Build;
+
+partial class Build
+{
+    [Parameter(Name = "CalamariFlavour")] readonly string CalamariFlavourToTest;
+    [Parameter(Name = "VSTest_TestCaseFilter")] readonly string CalamariFlavourTestCaseFilter;
+
+    [PublicAPI]
+    Target TestCalamariFlavourProject =>
+        target => target
+            .Executes(async () =>
+                      {
+                          var testProject = $"Calamari.{CalamariFlavourToTest}.Tests";
+
+                          var affectedProjectFile = RootDirectory / "affected.proj";
+                          bool isAffected;
+                          if (affectedProjectFile.FileExists())
+                          {
+                              Log.Information("Affected projects analysis found; checking to see if {TestProject} is affected", testProject);
+                              var contents = await File.ReadAllTextAsync(affectedProjectFile);
+
+                              isAffected = contents.Contains(testProject);
+                          }
+                          else
+                          {
+                              Log.Information("Affected projects analysis not found; assuming {TestProject} *is* affected", testProject);
+                              isAffected = true;
+                          }
+
+                          // Force test to ensure master build will work
+                          if (isAffected || true)
```

Comment:
This will be removed before merging, it is just to run once to ensure tests still run correctly on master